### PR TITLE
Provide serialization/deserialization of multiple configuration values.

### LINKED ISSUE #4641: Configuration Parser do not support multiple values
There is no support for multiple values in Automation  ConfigurationSerializer and ConfigurationDeserializer

### PR DESCRIPTION
**Pull request by adimova**
Closes #4641

Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java [hunk 1]
```diff
@@ -38,16 +41,39 @@ public Configuration deserialize(JsonElement json, Type typeOfT, JsonDeserializa
...
+    private void consume(String key, JsonPrimitive value, Configuration configuration) {
+        Object res = null;
+        if (value.isString()) {
+           res = value.getAsString();
+        } else if (value.isNumber()) {
+            res = value.getAsBigDecimal();
+        } else if (value.isBoolean()) {
+            res = value.getAsBoolean();
+        }
+        if (res == null) return;
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java]: At least this line looks like your IDE is not working correctly. The save action should trigger the formatter and the formatter should change this line.

**Comment by adimova** [comment]
`org.eclipse.smarthome.storage.json.internal.ConfigurationDeserializer` is moved to `org.eclipse.smarthome.config.core` package and is modified to conform also to the automation json format of the configuration

**Comment by maggu2810** [comment]
@kaikreuzer Do you still request some changes?

**Comment by kaikreuzer** [comment]
Sorry, I lost that out of sight.
The PR looks good to me, but it feels kind of weird to have `ConfigurationDeserializer` in config.core, but `ConfigurationSerializer` still in automation.parser. Wouldn't it make sense to move that class as well (even if currently not being used by anyone else than the automation component, but just for the sake of symmetry)?

**Comment by maggu2810** [comment]
@adimova I assume if you use history rewriting and force push your changes we don't get any mail notification (at least I didn't receive one). So it would be nice, if you add a new commit or better add a comment that the requested changes has been fixed.

**Comment by adimova** [comment]
Sorry, of course I will. The `ConfigurationSerializer` is in config.core already.

**Comment by maggu2810** [comment]
With the latest merges there are conflicts again.
@kaikreuzer as @adimova moved the serializer class to the other package are there still some changes to be done or could we merge this after the conflicts has been solved again (and before new ones)?

**Comment by kaikreuzer** [comment]
No, nothing open from my side and yes, sorry that I missed that a commit had been pushed to the branch.

@adimova May I thus ask you for a final rebase and I promise to merge immediately after? Thanks!

**Comment by adimova** [comment]
done

**Comment by kaikreuzer** [comment]
Done as well, thanks!

**Pull request merged**